File: kalitech-engine/src/main/resources/kalitech/engine/modules/Render/Render.js
// FILE: resources/kalitech/engine/modules/Render/Render.js
// Author: Kalitech
"use strict";

const {requireEngineApi, normalizeCfgObject} = require("../helpers/ModuleCommon.js");

/**
 * Render API wrapper.
 * Provides simple methods that map directly to RenderApi contract.
 */
function create(engine /*, K */) {
    const api = requireEngineApi(engine, "render", "RENDER");

    return Object.freeze({
        /** Ensure render scene is initialized. */
        ensureScene() {
            api.ensureScene();
            return true;
        },

        /** Configure ambient light. */
        ambient(cfg) {
            api.ambientCfg(normalizeCfgObject(cfg));
            return true;
        },

        /** Configure sun (directional) light. */
        sun(cfg) {
            api.sunCfg(normalizeCfgObject(cfg));
            return true;
        },

        /** Configure sun shadows. */
        sunShadows(cfg) {
            api.sunShadowsCfg(normalizeCfgObject(cfg));
            return true;
        },

        /** Configure fog. */
        fog(cfg) {
            api.fogCfg(normalizeCfgObject(cfg));
            return true;
        },

        /** Configure post-processing. */
        post(cfg) {
            api.postCfg(normalizeCfgObject(cfg));
            return true;
        },

        /** Access to the raw host API. */
        api
    });
}

create.META = {
    moduleId: "render",
    globalName: "RENDER",
    version: "1.0.0",
    description: "Render wrapper for scene setup, lighting, fog and post-processing",
    engineMin: "0.1.0"
};

module.exports = create;

File: modulesSrc/RenderModule/src/main/resources/kalitech/engine/modules/Render/Render.lua
local M = {}
local lua_require_result_0 = require("@builtin/modules/helpers/ModuleCommon.lua")
requireEngineApi = lua_require_result_0.requireEngineApi
normalizeCfgObject = lua_require_result_0.normalizeCfgObject
create = setmetatable(
    {},
    {__call = function(lua_, self, engine)
        local api = requireEngineApi(_G, engine, "render", "RENDER")
        return KObject:freeze({
            ensureScene = function(self)
                api:ensureScene()
                return true
            end,
            ambient = function(self, cfg)
                api:ambientCfg(normalizeCfgObject(_G, cfg))
                return true
            end,
            sun = function(self, cfg)
                api:sunCfg(normalizeCfgObject(_G, cfg))
                return true
            end,
            sunShadows = function(self, cfg)
                api:sunShadowsCfg(normalizeCfgObject(_G, cfg))
                return true
            end,
            fog = function(self, cfg)
                api:fogCfg(normalizeCfgObject(_G, cfg))
                return true
            end,
            post = function(self, cfg)
                api:postCfg(normalizeCfgObject(_G, cfg))
                return true
            end,
            api = api
        })
    end}
)
create.META = {
    moduleId = "render",
    version = "1.0.0",
    description = "Render wrapper for scene setup, lighting, fog and post-processing",
    engineMin = "0.1.0"
}
M = create

return M
